### ISSUE TITLE
Early Stopping

### ISSUE THREAD

**Issue by ragavendranbala** [rajat-1994/realtime-Image-segmentation-onmobile#2]
Epoch 00015: val_loss did not improve from 0.06950
869/869 [==============================] - 395s 454ms/step - loss: 0.0615 - IOU: 0.9218 - val_loss: 0.0707 - val_IOU: 0.9160 - lr: 5.0000e-04
Epoch 16/50
869/869 [==============================] - ETA: 0s - loss: 0.0609 - IOU: 0.9224
Epoch 00016: val_loss did not improve from 0.06950

Epoch 00016: ReduceLROnPlateau reducing learning rate to 0.0002500000118743628.
869/869 [==============================] - 405s 466ms/step - loss: 0.0609 - IOU: 0.9224 - val_loss: 0.0703 - val_IOU: 0.9157 - lr: 5.0000e-04
Epoch 17/50
869/869 [==============================] - ETA: 0s - loss: 0.0594 - IOU: 0.9241
Epoch 00017: val_loss improved from 0.06950 to 0.06806, saving model to ./weights/hair_segmentation_mobile.h5
869/869 [==============================] - 408s 469ms/step - loss: 0.0594 - IOU: 0.9241 - val_loss: 0.0681 - val_IOU: 0.9193 - lr: 2.5000e-04
Epoch 18/50
869/869 [==============================] - ETA: 0s - loss: 0.0591 - IOU: 0.9245
Epoch 00018: val_loss did not improve from 0.06806
869/869 [==============================] - 405s 466ms/step - loss: 0.0591 - IOU: 0.9245 - val_loss: 0.0737 - val_IOU: 0.9132 - lr: 2.5000e-04
Epoch 19/50
869/869 [==============================] - ETA: 0s - loss: 0.0587 - IOU: 0.9250
Epoch 00019: val_loss did not improve from 0.06806
869/869 [==============================] - 407s 469ms/step - loss: 0.0587 - IOU: 0.9250 - val_loss: 0.0713 - val_IOU: 0.9164 - lr: 2.5000e-04
Epoch 20/50
869/869 [==============================] - ETA: 0s - loss: 0.0587 - IOU: 0.9250
Epoch 00020: val_loss did not improve from 0.06806

Epoch 00020: ReduceLROnPlateau reducing learning rate to 0.0001250000059371814.
869/869 [==============================] - 401s 461ms/step - loss: 0.0587 - IOU: 0.9250 - val_loss: 0.0684 - val_IOU: 0.9175 - lr: 2.5000e-04
Epoch 21/50
869/869 [==============================] - ETA: 0s - loss: 0.0580 - IOU: 0.9258
Epoch 00021: val_loss did not improve from 0.06806
869/869 [==============================] - 404s 465ms/step - loss: 0.0580 - IOU: 0.9258 - val_loss: 0.0681 - val_IOU: 0.9187 - lr: 1.2500e-04
Epoch 22/50
869/869 [==============================] - ETA: 0s - loss: 0.0577 - IOU: 0.9262
Epoch 00022: val_loss did not improve from 0.06806
869/869 [==============================] - 426s 490ms/step - loss: 0.0577 - IOU: 0.9262 - val_loss: 0.0685 - val_IOU: 0.9179 - lr: 1.2500e-04
Epoch 00022: early stopping

It happened 3rd Time(Used Ryzen 7 + GTX 1060), Only one time it crossed 30 and stopped on i9 CPU

**Comment by rajat-1994** [comment]
I dont see any problem here. Can you tell me exactly what is the problem?

**Comment by ragavendranbala** [comment]
It's showing iterations as 50 but stopping at 20 itself? and also I'm facing some accuracy issues, Can you tell me how to increase the accuracy?

**Comment by rajat-1994** [comment]
50 is the maximum number of epochs and it is stopping because validation loss is not decreasing anymore, early stopping is a way to stop your model from early and not train unnecessary which can lead to overfitting.

You can increase the accuracy by increasing input size of  the model maybe 224x224,256x256 ,but this will slow the model.

**Comment by ragavendranbala** [comment]
@rajat-1994  Got it,Thanks, Have you tried Mediapipe Hair Segmentation models without custom Ops? Does it give Good Accuracy?

**Comment by rajat-1994** [comment]
Sorry I have not benchmarked it

**Comment by ragavendranbala** [comment]
Don't know to remove custom_ops from mediapipe so confused. if possible help me I'm closing this
